nextcord/http.py

```diff
@@ -151,48 +471,172 @@ def __exit__(
 
 
 class HTTPClient:
-    """Represents an HTTP client sending HTTP requests to the Discord API."""
+    """Represents an HTTP client sending HTTP requests to the Discord API.
+
+    Also, not thread safe.
+
+    Parameters
+    ----------
+    connector: Optional[:class:`aiohttp.BaseConnector`]
+        The connector object to use for the client session.
+    max_global_requests: :class:`int`
+        Maximum amount of requests per second per authorization.
+
+        Discord by default only allows 50 requests per second, but if your bot has had its maximum increased, then
+        increase this parameter.
```

Comment:
Since you have access to the gateway parts, why not make this estimate based on the guild/shard count?
Might not be accurate, but worst that can happen is they get less perf than what they could if they aren't manually setting it.